node/pkg/watchers/algorand/watcher.go

```diff
@@ -237,7 +238,7 @@ func (e *Watcher) Run(ctx context.Context) error {
 		case <-ctx.Done():
 			return nil
 		case r := <-e.obsvReqC:
-			if vaa.ChainID(r.ChainId) != vaa.ChainIDAlgorand {
+			if r.ChainId > math.MaxUint16 || vaa.ChainID(r.ChainId) != vaa.ChainIDAlgorand {
```

Comment:
When I saw this, my first thought was that someone could exploit this to cause a panic. However, an event is only posted on this channel in `reobserve.go` if `r.ChainId` is for this chain so that can't happen. Just wonder if that's worth a comment here (and in the other watchers) to keep others from wondering the same thing?

Reply:
Just FYI this type of thing motivated this PR: https://github.com/wormhole-foundation/wormhole/pull/4271

Reply:
Added comments to the watchers